### PR TITLE
[JENKINS-34287,JENKINS-23232] Update the component towards 1.34

### PR DESCRIPTION
**Pull request by oleg-nenashev**
It is required to release the fix for #6

- [x] - Update Parent POM
- [x] - Fixed some serious FindBugs errors (stuff like DM_DEFAULT_ENCODING is left)

@reviewbybees @daniel-beck

### CODE REVIEW

**Comment by svanoort** [comment]
🐝  LGTM

File: src/main/java/Main.java [hunk 1]
```diff
@@ -194,7 +204,7 @@ private static void _main(String[] args) throws Exception {
         // is deployed.
         File tempFile = File.createTempFile("dummy", "dummy");
         deleteContents(new File(tempFile.getParent(), "winstone/" + me.getName()));
-        tempFile.delete();
+        Files.delete(tempFile.toPath());
```
Comment (oleg-nenashev) [inline src/main/java/Main.java]: Maybe better to suppress it

**Comment by daniel-beck** [comment]
> Hop from Java 1.3 to Java 7 as a new target

Defeats the purpose of most of https://github.com/jenkinsci/extras-executable-war/blob/5a5f01362e7649da3b39a3a45940ba3ce1ab65e8/src/main/java/Main.java#L78...L96 – maybe not 1.3 as target, but could we do 1.5 or 1.6?

**Comment by oleg-nenashev** [comment]
> maybe not 1.3 as target, but could we do 1.5 or 1.6?

Well, we could. Just requires rework of FindBugs fixes

**Comment by rsandell** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
Downgrading to Java 6

File: src/main/java/Main.java [hunk 2]
```diff
@@ -71,15 +75,18 @@
         if (dependenciesInputStream == null) {
             throw new IOException("Cannot find resource " + DEPENDENCIES_LIST);
         }
-        Map<String,String> r = new HashMap<>();
-        try (BufferedReader in = new BufferedReader(new InputStreamReader(dependenciesInputStream))) {
+        final Map<String,String> r = new HashMap<String,String>();
+        final BufferedReader in = new BufferedReader(new InputStreamReader(dependenciesInputStream));
```
Comment (rsandell) [inline src/main/java/Main.java]: Should be moved inside the try block

Comment (oleg-nenashev) [inline src/main/java/Main.java]: Are you aware about RuntimeException?

**Comment by daniel-beck** [comment]
@oleg-nenashev Thanks. What are the reasons to go from 1.3 to any of the newer versions (to document for posterity why we did that at all)?

**Comment by oleg-nenashev** [comment]
@daniel-beck 
> What are the reasons to go from 1.3 to any of the newer versions (to document for posterity why we did that at all)?

No specific reasons excepting the fact I was unable to compile it on JDK8. But it compiles well on JDK7. So I've downgraded it

**Comment by stephenc** [comment]
Target 1.7 enables faster classloading... be good if we could get every jar used in Jenkins compiled with target 1.7

**Comment by daniel-beck** [comment]
@stephenc 

> Target 1.7 enables faster classloading... be good if we could get every jar used in Jenkins compiled with target 1.7

Do these handful of classes, with a very specific reason to use a low target JRE, impact this in any significant way? E.g. it's only fast if all classes are 1.7+?

---

@oleg-nenashev 

> No specific reasons excepting the fact I was unable to compile it on JDK8. But it compiles well on JDK7. So I've downgraded it

And JDK 7 requires 1.6+?

**Comment by oleg-nenashev** [comment]
I propose to keep the 1.3 compatibility since we already have this PR adapted. Then we can go after 1.7 as a bulk change in core modules

**Pull request merged**